pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -74,7 +75,7 @@ func (pl *PodTopologySpread) initPreScoreState(s *preScoreState, pod *v1.Pod, fi
 	}
 	topoSize := make([]int, len(s.Constraints))
 	for _, node := range filteredNodes {
-		if !nodeLabelsMatchSpreadConstraints(node.Labels, s.Constraints) {
+		if !isSystemDefaultConstraints(pod, pl.defaultConstraintsType) && !nodeLabelsMatchSpreadConstraints(node.Labels, s.Constraints) {
```

Comment:
isSystemDefaultConstraints() should be called once, rather than being called N time inside the loop.